tools/enable_local_firecracker.sh

```diff
@@ -5,63 +5,62 @@ set -euo pipefail
 # user to run firecracker (via the jailer), without being root.
 
 # this script must be run as root, so check that first.
-if [ $(id -u) -ne 0 ]; then
+if [ "$(id -u)" -ne 0 ]; then
     echo "This script must be run as root (use sudo)."
     echo "sudo $0"
     exit 1
 fi
 
-if ! command -v jailer &> /dev/null
-then
+if ! command -v jailer &>/dev/null; then
     echo "jailer could not be found (install firecracker + jailer?)"
     exit 1
 fi
 
-if ! command -v ip &> /dev/null
-then
+if ! command -v ip &>/dev/null; then
     echo "ip could not be found (install iproute2?)"
     exit 1
 fi
 
-if ! command -v iptables &> /dev/null
-then
+if ! command -v iptables &>/dev/null; then
     echo "iptables could not be found"
     exit 1
 fi
 
+CGROUP2_PATH=$( (mount | grep -E -m1 '^cgroup2 on ' | awk '{print $3}') || true)
+if [[ ! "$CGROUP2_PATH" ]]; then
+    echo "missing cgroup2 mount"
+    exit 1
+fi
+
 groupadd -f -r cgroups
 usermod -a -G cgroups root
-usermod -a -G cgroups $SUDO_USER
+usermod -a -G cgroups "$SUDO_USER"
 
 # jailer will create stuff here; ensure the dir exists and owner is user.
-mkdir -p /sys/fs/cgroup/cpuset/firecracker
-chown -R $SUDO_USER:cgroups /sys/fs/cgroup/cpuset/firecracker
-chmod -R g+rw /sys/fs/cgroup/cpuset/firecracker
-
-mkdir -p /sys/fs/cgroup/firecracker
-chown -R $SUDO_USER:cgroups /sys/fs/cgroup/firecracker
-chmod -R g+rw /sys/fs/cgroup/firecracker
+mkdir -p "$CGROUP2_PATH"/firecracker
+chown -R "$SUDO_USER":cgroups "$CGROUP2_PATH"/firecracker
+chmod -R g+rw "$CGROUP2_PATH"/firecracker
 
-chown -R root:cgroups /sys/fs/cgroup/cgroup.subtree_control
-chmod -R g+rw /sys/fs/cgroup/cgroup.subtree_control
-chown -R root:cgroups /sys/fs/cgroup/cgroup.procs
-chmod -R g+rw /sys/fs/cgroup/cgroup.procs
+chown -R root:cgroups "$CGROUP2_PATH"/cgroup.subtree_control
+chmod -R g+rw "$CGROUP2_PATH"/cgroup.subtree_control
+chown -R root:cgroups "$CGROUP2_PATH"/cgroup.procs
+chmod -R g+rw "$CGROUP2_PATH"/cgroup.procs
```

Comment:
I know this is my fault, but this should probably be:
```suggestion
chown -R "$SUDO_USER":cgroups "$CGROUP2_PATH"/cgroup.subtree_control
chmod -R g+rw "$CGROUP2_PATH"/cgroup.subtree_control
chown -R "$SUDO_USER":cgroups "$CGROUP2_PATH"/cgroup.procs
chmod -R g+rw "$CGROUP2_PATH"/cgroup.procs
```
right?

Reply:
firecracker_test seems to pass either way :shrug: 

I guess it doesn't matter since our user gets added to the cgroups group, and so we inherit permissions from there?

Reply:
Fair enough.

Reply:
Changed to SUDO_USER just for consistency.